### PR TITLE
fix: provide meaningful error messages and add test

### PR DESCRIPTION
**Pull request by JadhavPoonam**
### Description

- Add missing test cases and patch for variations of resource type format
- Consistent language/messaging across CLI commands
- Rely on the `flag.args` to confirm that required arguments are passed correctly

### Links

Feedback that prompted this [PR](https://hashicorp.slack.com/archives/C0253EQ5B40/p1694462086447919) 

### PR Checklist

* [x] updated test coverage
* [ ] ~external facing docs updated~
* [x] appropriate backport labels added
* [ ] ~not a security concern~

### CODE REVIEW

**Review by wangxinyi7 — APPROVED** [review]
Thanks for putting things together! It looks good to me.

File: command/resource/apply/apply.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func (c *cmd) Run(args []string) int {
 		}
 		parsedResource = data
 	} else {
-		c.UI.Error("Flag -f is required")
+		c.UI.Error("Your argument format is incorrect: Flag -f with file path argument is required")
```
Comment (dhiaayachi) [inline command/resource/apply/apply.go:88]:
```suggestion
		c.UI.Error("Incorrect argument format: Flag -f with file path argument is required")
```

File: command/resource/apply/apply_test.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func TestResourceApplyInvalidArgs(t *testing.T) {
 		"missing required flag": {
 			args:         []string{},
 			expectedCode: 1,
-			expectedErr:  errors.New("Flag -f is required"),
+			expectedErr:  errors.New("Your argument format is incorrect: Flag -f with file path argument is required"),
```
Comment (dhiaayachi) [inline command/resource/apply/apply_test.go:81]:
```suggestion
			expectedErr:  errors.New("Incorrect argument format: Flag -f with file path argument is required"),
```

File: command/resource/list/list_test.go [hunk 3]
```diff
@@ -99,12 +99,12 @@ func TestResourceListInvalidArgs(t *testing.T) {
 		"nil args": {
 			args:         nil,
 			expectedCode: 1,
-			expectedErr:  errors.New("Please provide required arguments"),
+			expectedErr:  errors.New("Your argument format is incorrect: Must include resource type argument"),
```
Comment (dhiaayachi) [inline command/resource/list/list_test.go:102]:
```suggestion
			expectedErr:  errors.New("Incorrect argument format: Must include resource type argument"),
```

File: command/resource/list/list_test.go [hunk 4]
```diff
@@ -99,12 +99,12 @@ func TestResourceListInvalidArgs(t *testing.T) {
...
 		},
 		"minimum args required": {
 			args:         []string{},
 			expectedCode: 1,
-			expectedErr:  errors.New("Please provide required arguments"),
+			expectedErr:  errors.New("Your argument format is incorrect: Must include resource type argument"),
```
Comment (dhiaayachi) [inline command/resource/list/list_test.go:107]:
```suggestion
			expectedErr:  errors.New("Incorrect argument format: Must include resource type argument"),
```

**Review by dhiaayachi — APPROVED** [review]
Thank you @JadhavPoonam!
I left some wording suggestions to consider.

Also I'm wondering if this UX of accepting both combination of flags and print a warning is a better UX then making them mutually exclusive 🤔. I feel like accepting both and having one taking precedence could be a bit confusing. WYT?

**Comment by JadhavPoonam** [comment]
> > Thank you @JadhavPoonam! I left some wording suggestions to consider.
> > Also I'm wondering if this UX of accepting both combination of flags and print a warning is a better UX then making them mutually exclusive 🤔. I feel like accepting both and having one taking precedence could be a bit confusing. WYT?
> 
> Yeah, I tend to agree with you but I also don't want to punish the user by throwing an error when they provide say a `-f` arg with the inline args style.

Discussed this async and decided to throw an error instead of a warning in favor of simpler documentation.

**Pull request merged**